warp/localized.py

```
@@ -0,0 +1,765 @@
+        if buffer.nbytes == footprint_bytes and len(vmm_allocations) == 1:
+            # Single allocation with correct size, reinterpret bytes as dtype then reshape
+            # buffer is uint8, so we need to view it as the target dtype first
+            cupy_arr = buffer.view(np_dtype).reshape(global_shape)
+            print(f"✓ Using allocated cupy array with shape {global_shape}")
+            print(f"{'='*60}")
+            print()
+            return cupy_arr
+
+    # Multiple allocations or need to create unified view
+    # Fall back to managed memory for simplicity
+    print(f"Multiple allocations detected, creating unified managed memory view...")
```

Ruff (0.14.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)